pkg/server/commands/write.go

```diff
@@ -66,6 +68,10 @@ func (c *WriteCommand) validateWriteRequest(ctx context.Context, req *openfgapb.
 			return err
 		}
 
+		if IsAuthorizationModelObsolete(authModel.SchemaVersion, c.allowSchema10) {
+			return serverErrors.ObsoleteAuthorizationModel
+		}
```

Comment:
Every call I see to `IsAuthorizationModelObsolete` is followed by a `typesystem.New`  or `typesystem.NewAndValidate`.

Can we move this check to a new function `typesystem.NewAndValidateSchema`?